test/relayHub/RelayHub.test.ts

```diff
@@ -1,1177 +1,507 @@
+          ) => mockRelayHub.connect(manager).addRelayWorkers([workerAddr]);
 
-        const actualWorkerCount = await mockRelayHub.workerCount(
-          relayManagerAddr
-        );
+          beforeEach(async function () {
+            // Add stake to manager
+            await mockRelayHub.stakeForAddress(relayManagerAddr, 1000, {
+              value: oneRBTC,
+            });
 
-        expect(expectedWorkerCount).to.eq(actualWorkerCount);
-      });
+            // Add first worker for relayManager
+            await addWorker(relayManager);
 
-      it('should NOT be able to add workers when Manager is unstaked', async function () {
-        const promiseOfWorker = mockRelayHub
-          .connect(relayManager)
-          .addRelayWorkers([relayWorkerAddr]);
+            // Enable more than one worker per manager
+            await mockRelayHub.setVariable('maxWorkerCount', 2);
+          });
 
-        await expect(promiseOfWorker).to.have.revertedWith(
-          ERR_UNSTAKED
-        );
-      });
+          it('shoud map the manager addr onto the worker addr', async function () {
+            const expectedManagerAddr = relayManagerAddr;
+            const actualManager = await mockRelayHub.workerToManager(
+              relayWorkerAddr
+            );
+
+            const actualManagerAddr = ethers.utils.getAddress(
+              `0x${actualManager.slice(25, actualManager.length - 1)}`
+            );
+
+            expect(actualManagerAddr).to.eq(expectedManagerAddr);
+          });
+
+          it('should NOT register the same worker twice', async function () {
+            // Adding same worker for the same manager second time
+            await expect(addWorker(relayManager)).to.be.revertedWith(
+              ERR_WORKER_HAS_MANAGER
+            );
+          });
+
+          it('should NOT register the same worker for different managers', async function () {
+            const [anotherRelayManager] = otherUsers;
 
-      describe('with one worker', function() {
-        const addWorker = (
-          manager: SignerWithAddress,
-          workerAddr = relayWorkerAddr
-        ) => mockRelayHub.connect(manager).addRelayWorkers([workerAddr]);
+            await mockRelayHub.stakeForAddress(
+              anotherRelayManager.address,
+              1000,
+              {
+                value: oneRBTC,
+              }
+            );
 
-        beforeEach(async function() {
-          // Add stake to manager
+            await expect(addWorker(anotherRelayManager)).to.be.revertedWith(
+              ERR_WORKER_HAS_MANAGER
+            );
+          });
+
+          it('should fail when adding more workers than maxWorkerCount', async function () {
+            const expectedWorkerCountLimit = 13;
+            await mockRelayHub.setVariable('maxWorkerCount', expectedWorkerCountLimit);
+
+            await mockRelayHub.stakeForAddress(relayManagerAddr, 1000, {
+              value: oneRBTC.mul(expectedWorkerCountLimit),
```

Comment:
This .mul is not required, oneRBTC is enough (same in line 161)

Reply:
Done